bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Challenge/TallyChallenge.cs

```diff
@@ -0,0 +1,172 @@
+        InternalManifest manifest)
+    {
+        TallyId = tallyId;
+        GuardianId = guardian.GuardianId;
+        SequenceOrder = guardian.SequenceOrder;
+        Coefficient = coefficient;
+        Contests = manifest.ToContestChallengeDictionary();
+    }
+
+    public TallyChallenge(
+        string tallyId,
+        string guardianId,
+        LagrangeCoefficient coefficient,
+        InternalManifest manifest)
+    {
+        TallyId = tallyId;
+        GuardianId = guardianId;
+        SequenceOrder = coefficient.SequenceOrder;
+        Coefficient = coefficient.Coefficient;
+        Contests = manifest.ToContestChallengeDictionary();
+    }
+
+    public TallyChallenge(
+        string tallyId,
+        IElectionGuardian guardian,
+        LagrangeCoefficient coefficient,
+        Dictionary<string, ContestChallenge> contests)
+    {
+        if (guardian.SequenceOrder != coefficient.SequenceOrder)
+        {
+            throw new ArgumentException(
+                $"Guardian sequence order {guardian.SequenceOrder} does not match coefficient sequence order {coefficient.SequenceOrder}");
+        }
+        TallyId = tallyId;
+        GuardianId = guardian.GuardianId;
+        SequenceOrder = guardian.SequenceOrder;
+        Coefficient = new(coefficient.Coefficient);
+        Contests = contests
+            .ToDictionary(
+                kvp => kvp.Key,
+                kvp => new ContestChallenge(kvp.Value));
+    }
+
+    public TallyChallenge(TallyChallenge other) : base(other)
+    {
+        TallyId = other.TallyId;
+        GuardianId = other.GuardianId;
+        SequenceOrder = other.SequenceOrder;
+        Coefficient = new(other.Coefficient);
+        Contests = other.Contests
+            .ToDictionary(
+                kvp => kvp.Key,
+                kvp => new ContestChallenge(kvp.Value));
+    }
+
+    /// <summary>
+    /// Add an existing Selection challenge for a contest to the tally challenge
+    /// </summary>
+    public void Add(
+        IElectionContest contest,
+        SelectionChallenge selection)
+    {
+        Contests[contest.ObjectId].Add(selection);
+    }
+
+    /// <summary>
+    /// Add a new Selection challenge for a contest to the tally challenge
+    /// </summary>
+    public void Add(
+        IElectionContest contest,
+        IElectionSelection selection,
+        ElementModQ challenge)
+    {
+        Contests[contest.ObjectId].Add(
+            new SelectionChallenge(
+                selection.ObjectId,
+                GuardianId,
+                SequenceOrder,
+                Coefficient,
+                challenge
+            )
+        );
+    }
+
+    protected override void DisposeManaged()
+    {
+        base.DisposeManaged();
+        Contests.Dispose();
+    }
+
+    protected override void DisposeUnmanaged()
+    {
+        base.DisposeUnmanaged();
+        Coefficient?.Dispose();
+    }
+}
+
+public static partial class InternalManifestExtensions
```

Comment:
yeah i had considered that but left them like this for a few reasons.
- they are really only used in these subsegments (so most of them could probably just be private static members inside the classes)
- in one of the other comments i mentioned we should consider making a common abstract base class, so i didnt want to move it before we decide to do that or not
- we are intending to do some refactoring soon, which is a good time to look at all of these types of things holistically and figure out where they belong (i suspect we'll land on encapsulating them all at the module level rather than the feature level)